### PR TITLE
KTOR-1406 Throw ParseException on invalid header and handle it as 400

### PR DESCRIPTION
**Pull request by rsinukov**
**Subsystem**
Server

**Motivation**
https://youtrack.jetbrains.com/issue/KTOR-1406

### CODE REVIEW

**Comment by e5l** [comment]
There is a compile error on CI, please check

**Comment by rsinukov** [comment]
@e5l thanks, updated.

**Review by e5l — APPROVED** [review]
The implementation LGTM.
This PR can affect Space, @cy6erGn0m wdyt?

File: ktor-http/common/src/io/ktor/http/parsing/ParseException.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.http.parsing
+
+public class ParseException(
```
Comment (cy6erGn0m) [inline ktor-http/common/src/io/ktor/http/parsing/ParseException.kt:7]: This should be either documented or internal. Also, please don't forget to run apiDump for this

**Review by cy6erGn0m — APPROVED** [review]

**Comment by rsinukov** [comment]
@e5l new `ParseException` extends `IllegalArgumentExcpetion` for backwards compatibility. If users were handling it, it should keep the behavior. Otherwise, we will return the proper 400 status instead of 500.

File: ktor-http/common/src/io/ktor/http/parsing/ParseException.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-http/common/src/io/ktor/http/parsing/ParseException.kt:13]: thanks, done

**Pull request merged**